src/jemalloc.c

```diff
@@ -1379,8 +1382,21 @@ malloc_conf_init_helper(sc_data_t *sc_data, unsigned bin_shard_sizes[SC_NBINS],
 				CONF_HANDLE_BOOL(opt_xmalloc, "xmalloc")
 			}
 			CONF_HANDLE_BOOL(opt_tcache, "tcache")
-			CONF_HANDLE_SSIZE_T(opt_lg_tcache_max, "lg_tcache_max",
-			    -1, (sizeof(size_t) << 3) - 1)
+			CONF_HANDLE_SIZE_T(opt_tcache_max, "tcache_max",
+			    0, TCACHE_MAXCLASS_LIMIT, CONF_DONT_CHECK_MIN,
+			    CONF_CHECK_MAX, /* clip */ false)
+			if (CONF_MATCH("lg_tcache_max")) {
+				size_t m;
```

Comment:
Two things:
- I think better is to compare m against TCACHE_LG_MAXCLASS_LIMIT (this will better handle cases where the user erroneously inputs a number larger than the bits in size_t).
- I think we should clip rather than erroring out in case the user exceeds 8MB (i.e. I think "cache as much as you can" is a reasonable thing for the user to request).

Reply:
Changed both.